backend/Alpaki.Logic/Features/Invitations/InviteAVolunteer/InviteAVolunteerHandler.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Alpaki.CrossCutting.Enums;
+using Alpaki.Database;
+using Alpaki.Database.Models.Invitations;
+using MediatR;
+
+namespace Alpaki.Logic.Features.Invitations.InviteAVolunteer
+{
+    public class InviteAVolunteerHandler : IRequestHandler<InviteAVolunteerRequest, InviteAVolunteerResponse>
+    {
+        private readonly IDatabaseContext _databaseContext;
+        private readonly InvitationUniqueCodesGenerator _generator;
+        private readonly IMediator _mediator;
+
+        public InviteAVolunteerHandler(IDatabaseContext databaseContext, InvitationUniqueCodesGenerator generator, IMediator mediator)
+        {
+            _databaseContext = databaseContext;
+            _generator = generator;
+            _mediator = mediator;
+        }
+        public async Task<InviteAVolunteerResponse> Handle(InviteAVolunteerRequest request, CancellationToken cancellationToken)
+        {
+            var existingInvitation = _databaseContext.Invitations.SingleOrDefault(x => x.Email.ToLower() == request.Email.ToLower() && x.Status == InvitationStateEnum.Pending);
```

Comment:
SingleOrDefaultAsync

Reply:
a co jak dany email już jest zaproszony? chyba nie chemy 2 wolontariuszy z takim samym mailem?

dodajmy w fluentValidatorze checka że taki już istnieje!

Reply:
> a co jak dany email już jest zaproszony? chyba nie chemy 2 wolontariuszy z takim samym mailem?

Nie dodałem tutaj, bo później przy akceptacji to jest sprawdzane, ale racja mogę dodać to jeszcze tutaj.